vm.go

```diff
@@ -718,12 +718,8 @@ func (vm *vm) saveCtx(ctx *vmContext) {
 }
 
 func (vm *vm) pushCtx() {
-	if len(vm.callStack) > vm.maxCallStackSize {
-		ex := &StackOverflowError{}
-		panic(&uncatchableException{
-			stack: &ex.stack,
-			err:   ex,
-		})
+	if vm.maxCallStackSize != 0 && len(vm.callStack)+1 >= vm.maxCallStackSize {
```

Comment:
[q] why do we need to align this condition with the depth one? The `panic` change looks great to me.

Reply:
Ended up keeping depth, so changed this back! I think the only difference is whether the stack can grow to 1 over the limit or 1 smaller than the limit, so happy to keep it in line with the upstream.